### PR TITLE
added scan -- deviation from sync library - returns a list instead of generator

### PR DESCRIPTION
**Pull request by debosmit**
To make it closer to what the sync library usage pattern is, I could do `raise gen.Return(iter(result))` which would make usage pattern something like,

```
scan_res  = yield scan....
results = list(scan_res)
```

### CODE REVIEW

File: tornado_elasticsearch.py [hunk 1]
```diff
@@ -229,6 +229,63 @@ class for serialization
...
+    """
+    if not preserve_order:
+        body = query.copy() if query else {}
+        body["sort"] = "_doc"
+    # initial search
+    resp = yield client.search(body=query, scroll=scroll, size=size,
+                               request_timeout=request_timeout, **kwargs)
+    result = []
+    try:
+        raise gen.Return(result)
```
Comment (debosmit) [inline tornado_elasticsearch.py]: Returning an iterable here would make it possible to maintain a usage pattern similar to the sync client.

Comment (debosmit) [reply tornado_elasticsearch.py]: Something like `raise gen.Return(iter(result))`

Comment (thomdixon) [reply tornado_elasticsearch.py]: I would return the `iter` just to maintain compatibility. 👍

**Comment by thomdixon** [comment]
Do you have tests for this? Or did you test it? Are they in the other diff?

**Pull request merged**